### PR TITLE
Fixed  restoration of query tool database connection after dropping and re-creating the database with same name. #6487

### PR DESCRIPTION
**Pull request by anilsahoo20**

### CODE REVIEW

**Review by nikhil-mohite — CHANGES_REQUESTED** [review]
1. The Query tool is not working if the database name contains special characters (', "). Also, check autocomplete with this.
2. Please check `Open in new tab` for the Query tool. (Check database with special characters like ',", /,\), it is throwing error `NOT FOUND`
3. SchemaDiff is also failing if we drop and re-create the database with the same name.

**Review by nikhil-mohite — CHANGES_REQUESTED** [review]
1. The Query tool is not working if the database name contains special characters (', "). Also, check autocomplete with this.
2. Please check Open in new tab for the Query tool. (Check database with special characters like ',", /,), it is throwing error NOT FOUND
3. SchemaDiff is also failing if we drop and re-create the database with the same name.

**Review by nikhil-mohite — CHANGES_REQUESTED** [review]

File: web/pgadmin/tools/sqleditor/__init__.py [hunk 1]
```diff
@@ -432,7 +433,7 @@ def _init_sqleditor(trans_id, connect, sgid, sid, did, **kwargs):
         return True, internal_server_error(errormsg=str(e)), '', ''
 
     try:
-        conn = manager.connection(did=did, conn_id=conn_id,
+        conn = manager.connection(did=did, database=dbname, conn_id=conn_id,
```
[the remark this answers is not in the document]
Comment (anilsahoo20) [reply web/pgadmin/tools/sqleditor/__init__.py]: Made the changes

File: web/pgadmin/tools/schema_diff/__init__.py [hunk 2]
```diff
@@ -750,17 +750,12 @@ def get_schemas(sid, did):
     This function will return the list of schemas for the specified
     server id and database id.
     """
```
Comment (khushboovashi) [inline web/pgadmin/tools/schema_diff/__init__.py:752]: Any specific reason to remove exception handeling?

**Review by khushboovashi — CHANGES_REQUESTED** [review]
Please fix the below scenario:

1. Start a query tool and run select pg_sleep(50);
2. Now Delete the database and Recreate it.
3. Click continue on the warning message, it will start the execution.
4. Cancel the query through tool bar. 

It will throw No database Found error.

File: web/pgadmin/tools/schema_diff/__init__.py (hunk 2 above)
Comment (anilsahoo20) [reply web/pgadmin/tools/schema_diff/__init__.py]: There was one scenario given by Nikhil, which was basically giving issue in Browser tree schema collection object. The collection object was giving error on refresh of the node and Database connection gets disconnected, But Now if I test the same scenario, the issue is not reproducible and Also Nikhil confirmed that the issue is not reproducible after trying multiple times, so I am reverting to the earlier code.

**Pull request merged**